web/src/app/forms/FormField.js

```diff
@@ -82,17 +82,34 @@ export function FormField(props) {
     ..._inputProps,
   }
 
-  let getValueOf = (e) => (e && e.target ? e.target.value : e)
+  // mutable function
+  let getValueOf = (e) => {
+    if (e && e.target && e.target.value) {
+      return e.target.value
+    }
+    return e
+  }
+
   if (checkbox) {
     fieldProps.checked = fieldProps.value
     fieldProps.value = fieldProps.value ? 'true' : 'false'
-    getValueOf = (e) => e.target.checked
+    getValueOf = (e) => {
+      if (typeof e === 'string' || e instanceof Array) {
+        return e
+      }
+      return e.target.checked
+    }
   } else if (otherFieldProps.type === 'number') {
     fieldProps.label = label
     fieldProps.value = fieldProps.value.toString()
     fieldProps.InputLabelProps = InputLabelProps
-    getValueOf = (e) =>
-      float ? parseFloat(e.target.value) : parseInt(e.target.value, 10)
+    getValueOf = (e) => {
+      if (typeof e === 'string' || e instanceof Array) {
+        return e
+      }
```

Comment:
Same as above, use `isArray` and some context as to why we're mixing types. The `FormField` was intended to always operate on native types, instead of entered strings, (i.e., number if type=number)